### PR TITLE
Add the other specification of Git rev to Comparing commits

### LINKED ISSUE #19184: Some parts in "Comparing commits" do not work any more
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparisons-across-commits

### What part(s) of the article would you like to see updated?

In this page, now only "Comparisons across commits" asks a reader to type <rev> (commit sha1) on Web UI, but GitHub UI does not allow them to do so. You can remove that description:

```
As a shortcut, Git uses the ^ notation to mean "one commit prior."

You can use this notation to compare a single commit or branch against its immediate predecessors. For example, 96d29b7^^^^^ indicates five commits prior to 96d29b7, because there are five ^ marks. Typing 96d29b7^^^^^ in the base branch and 96d29b7 in the compare branch compares the five commits made before 96d29b7 with the 96d29b7 commit.
```

- 08ca047c87625c6706032f2e8ee668cea60ab32a (part of #17932) removes "type <rev>" from other section(s) on the page.
- Even as of disclosure of github/docs at 3df90fc9b811728a9b7ee45f598f9ae196783b54, there was already this explanation.

### Additional information

Current source code is available at:

https://github.com/github/docs/blob/94b4e599554687fc1d147d053cf528290819a918/content/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits.md?plain=1#L58-L64

### LINKED ISSUE #19185: Description in revision selection in "Comparing commits" is inadequate
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparisons-across-commits

### What part(s) of the article would you like to see updated?

In this page, "Comparisons across commits" section describes `^^^^^`, which is however equivalent to `~5`. Also the page misses the other specification `~` itself.

### Additional information

- [Ancestry References (in 7.1 Git Tools - Revision Selection)](https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection)

### Intra-repo information
- Relevant tissue: #19184

### PR DESCRIPTION
**Pull request by tnir**
### Why:

Closes #19184
Closes #19185

### What's being changed (if available, include any code snippets, screenshots, or gifs):

The section describes both ancestry references https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [n/a] This pull request impacts the contribution experience
  - [n/a] I have added the 'writer impact' label
  - [n/a] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

Signed-off-by: Takuya Noguchi [takninnovationresearch@gmail.com](https://github.com/sponsors/tnir)

### CODE REVIEW

**Comment by cmwilson21** [comment]
@tnir Thanks so much for opening a PR and linking it to your issue! I'll get this triaged for review :zap:

**Comment by cmwilson21** [comment]
SME  👍

**Comment by tnir** [comment]
@cmwilson21 Any updates on this?

File: content/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits.md [hunk 1]
```diff
@@ -57,12 +57,16 @@ Here's an example of a [comparison between two repositories](https://github.com/
...
 
-You can use this notation to compare a single commit or branch against its immediate predecessors. For example, `96d29b7^^^^^` indicates five commits prior to `96d29b7`, because there are five `^` marks. Typing `96d29b7^^^^^` in the `base` branch and `96d29b7` in the `compare` branch compares the five commits made before `96d29b7` with the `96d29b7` commit.
+You can use this notation to compare a single commit or branch against its immediate predecessors. For example, `96d29b7^^^^^` indicates five commits prior to `96d29b7`, because there are five `^` marks.
 
 Here's an example of a [comparison using the `^` notation](https://github.com/octocat/linguist/compare/octocat:96d29b7%5E%5E%5E%5E%5E...octocat:96d29b7).
 
+It's hard to archive such things as above as counting the number of `^` is an error-prone. Like ancestry references of the revision in Git, you can use the other notation `96d29b7~5`, which is equivalent to `96d29b7^^^^^`.
+
+Here's another example of a [comparison using the `~` notation](https://github.com/octocat/linguist/compare/octocat:96d29b7%7E5...octocat:96d29b7) for the same purpose as the above.
+
```
Comment (mattpollard) [inline content/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits.md:69]: Our table covers these concepts now, so this'll clean things up.

```suggestion
```

**Review by mattpollard — COMMENTED** [review]
@tnir 👋🏻 Thanks so much for contributing to GitHub Docs, and for your patience! We appreciate your suggestions. I left a few comments that consolidate your explanation and examples, with the goal of making this more concise.

Could you double-check that I haven't unintentionally introduced inaccuracies or lost any nuance, and let me know when you're ready for us to take another look? 🙏🏻

**Comment by cmwilson21** [comment]
👋 @tnir - Just wanted to check in and see if you've have had a chance to look over these suggestions 👀

**Review by mattpollard — APPROVED** [review]
Thanks, @cmwilson21 🙏🏻 @tnir, I'll go ahead and incorporate my suggestions from https://github.com/github/docs/pull/19186#pullrequestreview-1092906057, and we'll merge this shortly. If you have feedback or want to improve the article further, let us know via a new issue or pull request.

Thanks for your contribution! ✨

**Pull request merged**